### PR TITLE
Feature/algorithm store parameters default values

### LINKED ISSUE #1219: [Feature Request] Indicate whether arguments are optional and have default values
**Problem description**
Some algorithm arguments have default values, but users of the UI will now have to fill out all arguments every time.

**Desired solution**
- [x] Save in algorithm store whether parameter is obligatory
- [x] Save in algorithm store the default value if any? Maybe not - this is difficult if default value in the algorithm is updated
  - [x] Yes we should save default value: this is much more convenient in the UI and in future changes such as submitting a zip of the code (#1366) this becomes easier to verify that it is up-to-date
- [x] Allow to use the default value in the UI when creating task
- [x] Enter default values in client
- [x] Modify UI to include default values in submitting new algorithm in store
- [x] ~~Include https://github.com/vantage6/v6-algorithm-template/issues/6~~ -> such a for loop is not possible with copier
- [x] Check that default value matches with the argument type
- [x] Ensure that it works with multi-dropdown select parameters where `None` is the default
- [x] Support for entering default lists of values in UI

### PR DESCRIPTION
**Pull request by bartvanb**
Fix #1219

### CODE REVIEW

File: vantage6-algorithm-store/vantage6/algorithm/store/resource/algorithm.py [hunk 1]
```diff
@@ -421,6 +421,17 @@ def post(self):
                                   'string_list', 'integer', 'integer_list', 'float',
                                   'float_list', 'boolean', 'json', 'column',
                                   'column_list', 'organization' or 'organization_list'
+                              has_default_value:
+                                type: boolean
+                                description: Whether the argument has a default
+                                  value. If true, the 'default_value' field must be
+                                  provided. Default is false.
+                              default_value:
```
Comment (frankcorneliusmartin) [inline vantage6-algorithm-store/vantage6/algorithm/store/resource/algorithm.py:429]: How to set the default value to `NULL`? Just leaving the `default_value` unset? I think it is also possible to explicitly set it to `None`? (in that case the docstring needs to be updated)

**Review by frankcorneliusmartin — APPROVED** [review]
OK for the Python part

**Comment by frankcorneliusmartin** [comment]
@wbaccinelli can you review the UI part?

**Comment by wbaccinelli** [comment]
> @wbaccinelli can you review the UI part?
@frankcorneliusmartin I'm also not very familiar with the UI code

File: vantage6-algorithm-store/vantage6/algorithm/store/resource/algorithm.py (hunk 1 above)
Comment (bartvanb) [reply vantage6-algorithm-store/vantage6/algorithm/store/resource/algorithm.py:429]: Indeed, just leaving out `default_value` will set it to NULL. I adapted the docstring

File: vantage6-algorithm-store/vantage6/algorithm/store/resource/schema/input_schema.py [hunk 2]
```diff
@@ -109,6 +111,16 @@ def validate_type(self, value):
...
 
+    @validates_schema
+    def validate_default_value(self, data, **kwargs):
+        """
+        Validate that the default value is present if has_default_value is True.
+        """
+        if data.get("default_value") and not data.get("has_default_value"):
+            raise ValidationError(
+                "Default value cannot be given if has_default_value is False"
+            )
```
Comment (bartvanb) [inline vantage6-algorithm-store/vantage6/algorithm/store/resource/schema/input_schema.py:122]: Frank's comment  made me realize that we need to add validation here that if a default value is set, it should match the type. E.g. if the argument type is an integer, and the default value is 'blah', an error should be raised here

**Comment by frankcorneliusmartin** [comment]
Merge conflict resolved.

File: vantage6-algorithm-store/vantage6/algorithm/store/resource/schema/input_schema.py (hunk 2 above)
Comment (bartvanb) [reply vantage6-algorithm-store/vantage6/algorithm/store/resource/schema/input_schema.py:124]: In the commits I did yesterday this is addressed

**Pull request merged**